test/other/test_pthread_strict.c

```diff
@@ -0,0 +1,39 @@
+#include <stdio.h>
+#include <stdbool.h>
+#include <unistd.h>
+#include <pthread.h>
+#include <emscripten/console.h>
+
+void _emscripten_thread_set_strongref(pthread_t thread);
+
+pthread_mutex_t ready_lock = PTHREAD_MUTEX_INITIALIZER;
+pthread_cond_t ready_cond = PTHREAD_COND_INITIALIZER;
+_Atomic bool ready = false;
+
+void* thread_main(void* arg) {
+  pthread_mutex_lock(&ready_lock);
+  if (!ready) {
+    pthread_cond_wait(&ready_cond, &ready_lock);
+  }
+  pthread_mutex_unlock(&ready_lock);
+  emscripten_outf("Hello from thread");
+  return NULL;
+}
+
+int main() {
+  pthread_t t;
+  pthread_create(&t, NULL, thread_main, NULL);
+  pthread_detach(t);
+  emscripten_outf("thread created, returning from main");
+
+  // Make sure the thread output doesn't appear until after the above message
+  pthread_mutex_lock(&ready_lock);
+  ready = true;
+  pthread_cond_signal(&ready_cond);
+  pthread_mutex_unlock(&ready_lock);
+
+  // This needed under node to make sure the process doesn't exit before the
+  // thread is done.
+  _emscripten_thread_set_strongref(t);
```

Comment:
Do we expect users to need to do this?

Reply:
No, this is just do that node doesn't exit while the background thread is running.  I could instead use a setTimeout loop instead I guess?

Reply:
I guess I was unaware that this was an issue. If the main thread exits and a background thread remains, then we need to keep the main thread alive in a special way, I guess, that is the underlying issue?

Reply:
Right, its a little confusing.   Under normal circumstance, node will exit when the main function exits.  Running threads are not enough to keep node running.   The runtime has not exited in this case, the module is still available to be used, but node needs a reason to stay alive otherwise will exit the process.   

There is no equilient to this one the web, and this program will basically run forever on the web even without `_emscripten_thread_set_strongref`.. because there is no matching concept of "event loop is empty so exit" on the web.

I agree this is quite ugly here, so i might try to find a better way to keep node alive here.

Reply:
I think the canonical way is to remove the `pthread_detach()` call and replace `_emscripten_thread_set_strongref()` with `pthread_join()`, unless I'm missing something (perhaps this test case should not block the main thread?).

Reply:
Hmm..there must be a reason I didn't do it that way... i'll need to go figure out why :)

Reply:
Ah yes, the point of this test was to verify that when main returns it doesn't kill all the running threads.  If the thread was already joined there would be no point.

I re-wrote the test without threads at all PTAL